src/dataset/impl_dataset.rs

```diff
@@ -929,7 +849,10 @@ where
         M: for<'c> Fit<ArrayView2<'c, F>, ArrayView2<'c, E>, ER, Object = O>,
         O: for<'d> PredictInplace<ArrayView2<'a, F>, ArrayBase<OwnedRepr<E>, I>>,
         FACC: Float,
-        C: Fn(&ArrayView1<E>, &ArrayView1<E>) -> std::result::Result<FACC, crate::error::Error>,
+        C: Fn(
+            &ArrayView2<E>,
+            &ArrayView2<E>,
+        ) -> std::result::Result<Array1<FACC>, crate::error::Error>,
```

Comment:
What I meant in my comment was that since `cross_validate` can be expressed in terms of `cross_validate_multi`, we should only have one cross validation API that behaves like `cross_validate_multi`. This means `cross_validate` should behave the same as the old `cross_validate_multi`, and it should return a 2D array instead of 1D.

Reply:
Actually old `cross_validate_multi` returns a 1d array, while the previous implementation of `cross_validate` returns a scalar

Reply:
The signature of `cross_validate_multi` on master is
```
pub fn cross_validate_multi<O, ER, M, FACC, C>(
        &'a mut self,
        k: usize,
        parameters: &[M],
        eval: C,
    ) -> std::result::Result<Array2<FACC>, ER>
```
The return is actually a 2D array

Reply:
Ok my bad, i got confused